### PR TITLE
tests: update test for both ngx_lua_module and stream_ngx_lua_module

### PR DESCRIPTION
**Pull request by rainingmaster**

### CODE REVIEW

File: t/Test.pm [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+$ENV{TEST_NGINX_MYSQL_PORT} ||= 3306;
+$ENV{TEST_NGINX_MYSQL_HOST} ||= '127.0.0.1';
+$ENV{TEST_NGINX_MYSQL_PATH} ||= '/var/run/mysql/mysql.sock';
+
+no_long_string();
+
+add_block_preprocessor(sub {
+    my $block = shift;
+
+    if ($ENV{TEST_SUBSYSTEM} eq "stream") {
```
Comment (xiaocang) [inline t/Test.pm:29]: You should first determine if the environment variable exists.

File: t/Test.pm [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        if (!defined $block->stream_server_config) {
+            $block->set_value("stream_server_config", $block->server_config);
+        }
+    } else {
+        if (!defined $block->http_config) {
+            $block->set_value("http_config", $block->global_config);
+        }
+        if (!defined $block->request) {
+            $block->set_value("request", <<'_END_');
```
Comment (xiaocang) [inline t/Test.pm:41]: Just quotes here, very short string.

File: t/Test.pm [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+GET /t
+_END_
+        }
+        if (!defined $block->config) {
+            $block->set_value("config", "location /t {\n" . $block->server_config . "\n}");
```
Comment (xiaocang) [inline t/Test.pm:46]: Consider using heredoc with multiple rows of string.

File: t/big.t [hunk 4]
```diff
@@ -7,27 +7,13 @@ BEGIN {
     }
 }
 
-use Test::Nginx::Socket::Lua @skip;
-use Cwd qw(cwd);
+use t::Test;
```
Comment (xiaocang) [inline t/big.t:10]: you forgot `@skip` variable here.

File: t/Test.pm [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+$ENV{TEST_NGINX_MYSQL_PATH} ||= '/var/run/mysql/mysql.sock';
+
+no_long_string();
+
+add_block_preprocessor(sub {
+    my $block = shift;
+
+    if ($ENV{TEST_SUBSYSTEM} eq "stream") {
+        if (!defined $block->stream_config) {
+            $block->set_value("stream_config", $block->global_config);
```
Comment (xiaocang) [inline t/Test.pm:31]: Why introduce a new `block->global_config` here, while repeating it in every test block.
If you just use it as a default value of `stream_config` or `http_config`, just use the local variable if `stream_config` or `http_config` doesn't exist. (and you don't need a global variable)

**Review by xiaocang — CHANGES_REQUESTED** [review]

**Comment by xiaocang** [comment]
PR title here shoud be `tests: ...` not `test: ...` according to http://openresty.org/en/git-workflow.html.

File: t/Test.pm [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+use warnings;
+
+use Test::Nginx::Socket::Lua::Stream -Base;
+use Cwd qw(cwd);
+
+my $pwd = cwd();
+my $HtmlDir = html_dir;
+
+our @EXPORT = qw($GlobalConfig);
+our $GlobalConfig = qq{
```
Comment (xiaocang) [inline t/Test.pm:13]: You don't need a global variable (`$GlobalConfig`) and export it anymore, just use a local variable like `my $default_config` is enough.

**Review by xiaocang — APPROVED** [review]

**Comment by rainingmaster** [comment]
Thanks @xiaocang

**Pull request merged**